OpenRA.Mods.Common/Warheads/DamageWarhead.cs

```diff
@@ -34,13 +35,13 @@ public static object LoadVersus(MiniYaml yaml)
 				: new Dictionary<string, int>();
 		}
 
-		public int DamageVersus(ActorInfo victim)
+		public int DamageVersus(Actor victim)
 		{
-			var armor = victim.Traits.GetOrDefault<ArmorInfo>();
-			if (armor != null && armor.Type != null)
+			var armor = victim.TraitsImplementing<Armor>().Where(a => !a.IsTraitDisabled && a.Info.Type != null);
+			foreach (var a in armor)
 			{
 				int versus;
```

Comment:
I've now wasted 4 hours trying to make this work with no success, so unless someone else comes up with some concrete, working code, there's no telling when this will be finished.